lib/async/redis/context/pipeline.rb

```diff
@@ -0,0 +1,44 @@
+module Async
+	module Redis
+		module Context
+
+			# This class accumulates commands and sends several of them in a single
+			# request, instead of sending them one by one.
+			class Pipeline
+				include Methods::Strings
+				include Methods::Keys
+				include Methods::Lists
+
+				def initialize(connection_pool)
+					@pool = connection_pool
+					@connection = connection_pool.acquire
+
+					# Each command is an array where the first element is the name of the
+					# command ('SET', 'GET', etc.) and the rest of elements are the
+					# parameters for that command.
+					# Ex: ['SET', 'some_key', 42].
+					@commands = []
+				end
+
+				# This method just accumulates the commands and their params.
+				def call(*args)
+					@commands << args
+				end
+
+				# Send to redis all the accumulated commands.
+				# Returns an array with the result for each command in the same order
+				# that they were added with .call().
+				def run
+					@connection.write_pipeline(@commands)
```

Comment:
Ah, I forgot about this. I focused on the main use case for this which is using the pipelines from the client with `client.pipelined { |c| ... }`. Using that there's no need to reset the commands because a new instance is created on every call to `.pipelined`.

Anyway, you're totally right, the commands should be reset in this class. I changed that and also added a test for it in `pipeline_spec.rb`.